functions/supplier/debezium-supplier/src/main/java/org/springframework/cloud/fn/supplier/debezium/DebeziumReactiveConsumerConfiguration.java

```diff
@@ -49,8 +51,8 @@
  * @author Christian Tzolov
  * @author Artem Bilan
  */
-@Configuration
-@EnableConfigurationProperties(DebeziumProperties.class)
+@AutoConfiguration(after = DebeziumEngineBuilderAutoConfiguration.class)
```

Comment:
Now we probably have to add `META-INF/spring/org.springframework.boot.autoconfigure.AutoConfiguration.imports` for this project.

To be honest I think all of our out-of-the-box functions have to be auto-configuration. Otherwise their "drop-n-use" purpose is lost in target projects.

Reply:
I see now that I have gone through all commits that we have added the imports file and this is in fact an AC now! Yay! 

@tzolov do you mind exploring removing the config class from the source app pom.xml?

Reply:
@onobc I added the META-INFO spring auto-config but I'm afraid i can not remove the config class configuration form the source app at the moment. 
This is an uncharted territory and the `app.configClass` is a compulsory argument for the `spring-cloud-dataflow-apps-generator-plugin`: 
```
Execution app-gen of goal org.springframework.cloud:spring-cloud-dataflow-apps-generator-plugin:1.0.7:generate-app failed: No key, method or field with name 'app.configClass' on line 25 
```
We have to raise and solve the issue in the plugin first to enable the auto-configuration.

Reply:
That's fully different story.
If an auto-config was this function works even if it is imported again by that source application, we should leave that part outside of this PR and revise it when we already merge that plugin back to this project.
And probably already after we move functions into their own project.
There we would revise all the functions to be auto-configured since that would be a great feature in non-stream apps case where we want to compose functions ourselves.

Reply:
I thought that may happen @tzolov . No worries. I agree w/ @artembilan on the timeline - no rush and it does not hurt anything to be in both places for now.